### PR TITLE
HP-291: add test ids to HTML

### PR DESCRIPTION
**Pull request by NikoHelle**
Added to "data-test-id" -attributes to some components so testing is easier and tests more reliable even if components are moved in the DOM.

### CODE REVIEW

**Review by akikoskinen — CHANGES_REQUESTED** [review]
There are two commits, the first introduces a ton of changes (adds Testcafe stuff), the second removes most of that (removes Testcafe stuff). Craft commits that are concise and don't contain temporary unneeded stuff.

**Comment by NikoHelle** [comment]
Oh, sorry! Didn't remember there were a previous commit with the removed stuff. Squashed the commits.

File: src/components/LogoutInfo.tsx [hunk 1]
```diff
@@ -8,10 +8,12 @@ const LogoutInfoComponent = (): React.ReactElement => {
...
   return (
-    <div className={styles['content-element']}>
-      <h3>Olet kirjautunut, {name}</h3>
+    <div className={styles['content-element']} data-test-id="logout-info">
+      <h3>
+        Olet kirjautunut, <span data-test-id="logout-user-name">{name}</span>
+      </h3>
       <div>
-        <Button translate="" onClick={logout}>
+        <Button translate="" onClick={logout} data-test-id="logout-info">
```
Comment (akikoskinen) [inline src/components/LogoutInfo.tsx:16]: There's the same test id, `logout-info`, in two elements. Isn't that a bad thing?

Comment (NikoHelle) [reply src/components/LogoutInfo.tsx]: Logout button is not used in tests, so I removed the test-id from it.

**Review by akikoskinen — APPROVED** [review]
👍 Could squash the two commits into one, so the second doesn't have to fix the first one.

**Pull request merged**